### PR TITLE
Refresh README

### PR DESCRIPTION
**Pull request by jin**
- Updated TOC and Features
- Made bzlmod instructions more prominent
- Moved linked to examples higher up in the README
- Upleveled BOM section

### CODE REVIEW

File: README.md
Comment (shs96c) [inline README.md:67]: We use Bazel 7-only features in our `MODULE.bazel`.

File: README.md
Comment (shs96c) [inline README.md:74]: Do we want to encourage people to use basic authentication when Bazel itself won't? A simpler mental model of "if it works with Bazel, it'll work when resolving" might be better to lean into, and that suggests not mentioning basic auth....

File: README.md
Comment (shs96c) [inline README.md:103]: Bazel 7.

**Review by shs96c — APPROVED** [review]
The bazel 6 -> 7 thing is important to fix before landing, but that's a minor edit and won't need another review :)  LGTM.

File: README.md
Comment (jin) [reply README.md]: Fair point. Removed the basic auth mention.

**Pull request merged**
